### PR TITLE
Fixes #2041 Constantly creating processes while checking packages

### LINKED ISSUE #2041: Constantly creating processes while checking packages
Our file system watchers are detecting any change and retriggering scans. These scans import pip, which creates .pyc files. These files trigger another scan.

We should filter the file changes that may trigger a scan.

### PR DESCRIPTION
**Pull request by zooba**
Fixes #2041 Constantly creating processes while checking packages
Checks whether the file change is a Python source or compiled file before retriggering package scan.

### CODE REVIEW

File: Python/Product/VSInterpreters/PipPackageManager.cs [hunk 1]
```diff
@@ -675,37 +675,50 @@ private sealed class Suppressed : IDisposable {
...
+            }
         }
 
         public void NotifyPackagesChanged() {
             _refreshIsCurrentTrigger.Change(100, Timeout.Infinite);
         }
 
         private void OnRenamed(object sender, RenamedEventArgs e) {
-            _refreshIsCurrentTrigger.Change(1000, Timeout.Infinite);
+            if (ModulePath.IsPythonFile(e.FullPath, false, true, false) ||
```
Comment (huguesv) [inline Python/Product/VSInterpreters/PipPackageManager.cs]: What if it's a folder that's renamed?  We need to trigger unconditionally?

File: Python/Product/VSInterpreters/PipPackageManager.cs [hunk 2]
```diff
@@ -675,37 +675,50 @@ private sealed class Suppressed : IDisposable {
...
+                ModulePath.IsPythonFile(e.OldFullPath, false, true, false)) {
+                _refreshIsCurrentTrigger.Change(1000, Timeout.Infinite);
```
Comment (huguesv) [inline Python/Product/VSInterpreters/PipPackageManager.cs]: I'm wondering if you need to protect against ObjectDisposedException when changing the timer.

Comment (zooba) [inline Python/Product/VSInterpreters/PipPackageManager.cs]: Probably for the best.

File: Python/Product/VSInterpreters/PipPackageManager.cs (hunk 1 above)
Comment (zooba) [inline Python/Product/VSInterpreters/PipPackageManager.cs]: I believe we'll get a notification for each file contained in the folder, but I'll check.

Comment (zooba) [inline Python/Product/VSInterpreters/PipPackageManager.cs]: Nope, just one for the directory. I'll add that check.

**Comment by huguesv** [comment]
👍

**Pull request merged**